pkg/virt-launcher/virtwrap/api/schema.go

```diff
@@ -658,30 +659,52 @@ type InterfaceTarget struct {
 	Managed string `xml:"managed,attr,omitempty"`
 }
 
+// Tentatively keep "Name" exported for if we privatize the field, "kubectl get vm" does not output IP.
 type Alias struct {
-	Name string `xml:"name,attr"`
+	Name        string
```

Comment:
Name can be private. It should not be set from outside.

Reply:
@rmohr, This has been the only pain point for me. 

To name the variable as `name` or `Name` has huge impact to `virt-handler`. I have been spending days try to understand this. I do not know the root-cause, only find the earliest problem state in the VMI creation process. 

Specifically, without any other code change in my current commit, only replace the `Alias::Name` to `Alias::name`, and the direct references to it, the issue (a big one) appears. 

The issue is in `virt-handler/vm.go:execute()`, the following line, when retrieving the `api.Domain` object from Informer, the Alias::name fields will be empty for all devices (if use Alias::Name, the fields contain the right values).
 - https://github.com/kubevirt/kubevirt/blob/525399de9e17c65a266b29692256d051ff0148e2/pkg/virt-handler/vm.go#L1472
 - https://github.com/kubevirt/kubevirt/blob/525399de9e17c65a266b29692256d051ff0148e2/pkg/virt-handler/vm.go#L1091

Reply:
@rmohr 
With help from @xpivarc, we realized that the JSON-marshalled `api.Domain` information being passed failed correctly marshal the `Alias` object's private members. 

So I added the customized JSON marshal/unmarshal functions for the `api.Aliax` struct, and corresponding unit test.